compiler/src/iree/compiler/Dialect/Util/IR/UtilInterfaces.td

```diff
@@ -1621,4 +1621,32 @@ def Util_SerializableAttrInterface :
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// IREE::Util::StoredModuleAttrInterface
+//===----------------------------------------------------------------------===//
+
+def Util_StoredModuleAttrInterface : AttrInterface<"StoredModuleAttrInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Util";
+
+  let description = [{
+    Interface allowing attributes to store a reference to an underlying readonly
+    module. Thread safety of accesses to the module are managed by the
+    attribute.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/[{
+        Returns the underlying module given the annotation site.
```

Comment:
Could you define what you mean by 'annotation site'? Maybe an example would help.